src/App.h

```diff
@@ -149,6 +150,14 @@ struct TemplatedApp {
                     ->writeHeader("Connection", "Upgrade")
                     ->writeHeader("Sec-WebSocket-Accept", secWebSocketAccept);
 
+                std::string_view secWebSocketProtocol = req->getHeader("sec-websocket-protocol");
+                if (secWebSocketProtocol.length()) {
+                    auto ret = behavior.supportSubprotocol.selectBest(secWebSocketProtocol);
```

Comment:
I think selectBest could simply select the first protocol in the list passed to us. Basically take sec-websocket-protocol and just trim it by first comma.

Reply:
Yes. But I think **a websocket server may provides some subprotocols for communicate with clients, only the clients who support one of these subprotocols asked by server will be accept.** Thus, the server should choose the first protocol in the list which is also existed in server's protocol list. Am I think too much or just totally wrong.... not  sure about that.